src/Common/src/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -165,6 +168,15 @@ protected override IEntityHandleObject CreateValueFromKey(EntityHandle handle)
             }
         }
 
+        private object ResolveModuleReference(ModuleReferenceHandle handle)
+        {
+            ModuleReference moduleReference = _metadataReader.GetModuleReference(handle);
+            // Referenced modules are stored without their extension (see CommandLineHelpers.cs), so we have to drop
```

Comment:
Could you move this part (the extension stripping) to `ILVerifyTypeSystemContext` (to the `ResolveModule` method)?

I think CommandLineHelpers.cs will have to eventually change not to conflate assembly simple names with file names. I would like `\Common\TypeSystem` to already have the shape we want it to keep. We will want this API to operate on file names.